react/Modal/ModalEffects.ts

```diff
@@ -0,0 +1,47 @@
+import { Theme, useTheme } from '@material-ui/core'
+
+import { getFlagshipMetadata, isFlagshipApp } from 'cozy-device-helper'
+
+import { useSetFlagshipUI } from '../hooks/useSetFlagshipUi/useSetFlagshipUI'
+
+const getTopBackground = (theme: Theme, cozyBar: Element | null): string =>
+  (cozyBar && getComputedStyle(cozyBar).getPropertyValue('background-color')) ||
+  theme.palette.background.paper
+
+const getTopTheme = (cozyBar: Element | null): 'light' | 'dark' =>
+  getFlagshipMetadata().immersive ||
+  (cozyBar && cozyBar.classList.contains('coz-theme-primary'))
+    ? 'light'
+    : 'dark'
+
+const useHook = (): void => {
+  const theme = useTheme()
+  const cozyBar = document.querySelector('.coz-bar-wrapper')
+  const sidebar = document.getElementById('sidebar')
+
+  useSetFlagshipUI(
+    {
+      bottomBackground: theme.palette.background.paper,
+      bottomTheme: 'dark',
+      topBackground: theme.palette.background.paper,
+      topTheme: 'dark'
+    },
+    {
+      bottomBackground: theme.palette.background[sidebar ? 'default' : 'paper'],
+      bottomTheme: getFlagshipMetadata().immersive ? 'light' : 'dark',
+      bottomOverlay: 'transparent',
+      topOverlay: 'transparent',
+      topBackground: getTopBackground(theme, cozyBar),
+      topTheme: getTopTheme(cozyBar)
+    },
```

Comment:
This code has similarities with the one on `DialogEffects.ts`. Maybe we can homogenize them later?

Reply:
Yes I wanted to do that at first. But <Modal/> is a depreciated component, and there are too many small differences with <Dialog/> in my opinion. I'm not comfortable in mutualising this, I'm fearing side-effects. By the way, <Modal/> is only utilised in a single cozy-application (`cozy-store`). Once we remove <Modal/> from `cozy-store`, I will be comfortable with deleting this hook entirely.